File: frontend/src/interface/integration-update-request.ts
export interface IntegrationUpdateRequest {
    id: string;
    service_type?: string;
    auth_method?: string;
    connection_name?: string;
    host?: string;
    port?: number;
    database_name?: string;
    username?: string;
    password?: string;
    kerberos_principal?: string;
    windows_domain?: string;
    extra_options?: string;
    autosync_on?: boolean;
};

File: backend/shared/nextplore_shared/contracts/nextplore_orchestrator_service/integration_update_request.py
from pydantic import BaseModel
from typing import Optional


class IntegrationUpdateRequest(BaseModel):
    id: str
    service_type: Optional[str] = None
    auth_method: Optional[str] = None
    connection_name: Optional[str] = None
    host: Optional[str] = None
    port: Optional[int] = None
    database_name: Optional[str] = None
    username: Optional[str] = None
    password: Optional[str] = None
    kerberos_principal: Optional[str] = None
    windows_domain: Optional[str] = None
    extra_options: Optional[str] = None
    autosync_on: Optional[bool] = None
